client/src/document.js

```diff
@@ -137,18 +137,34 @@ function SidebarLeaf({ title, content }) {
 
 function SidebarLeaflets({ node }) {
   return (
-    <details>
-      <summary>{node.title}</summary>
-      <ol>
-        {node.content.map(childNode => {
-          return (
-            <li key={childNode.uri}>
-              <Link to={childNode.uri}>{childNode.title}</Link>
-            </li>
-          );
-        })}
-      </ol>
-    </details>
+    <Location>
+      {({ location }) => {
+        const activeChild = node.content.find(
```

Comment:
I'd probably move the `node.content.map`-ing up here and set the `hasActiveChild` inside of it, to minimize looping. It's a bit of a micro-optimization in this context, but it's good practice. e.g.

```jsx
let hasActiveChild = false;
const listItems = node.content.map(childNode => {
  const isActive = childNode.uri == location.pathname;
  if (isActive && !hasActiveChild) {
    hasActiveChild = true;
  }
  return <li/> // simplified
});
// rest of render
```

Reply:
But what if the link appears twice in the side bar?

Reply:
Applied! Thanks for the suggestion!